web/packages/teleterm/src/ui/TopBar/Identity/IdentityList/IdentityListItem.tsx

```diff
@@ -15,85 +15,157 @@
+  const profileColor = useStoreSelector(
+    'workspacesService',
+    useCallback(
+      state => state.workspaces[props.cluster.uri]?.color,
+      [props.cluster.uri]
+    )
+  );
 
   return (
-    <ListItem
-      css={`
-        border-radius: 0;
-        height: auto;
-        min-height: 38px;
-      `}
+    <StyledListItem
       onClick={props.onSelect}
+      onKeyDown={e =>
+        (e.key === 'Enter' || e.key === 'Space') && props.onSelect()
+      }
       isActive={isActive}
-      onMouseEnter={() => {
-        setIsHovered(true);
-      }}
-      onMouseLeave={() => {
-        setIsHovered(false);
-      }}
+      title={`Switch to ${props.cluster.name}`}
     >
-      <Flex justifyContent="space-between" alignItems="center" width="100%">
-        <Flex flexDirection="column" minWidth="0">
-          <Text typography="body2" title={userWithClusterName}>
-            {userWithClusterName}
-          </Text>
-          {props.cluster.profileStatusError && (
-            <ProfileStatusError
-              error={props.cluster.profileStatusError}
-              mb={1}
-            />
-          )}
-        </Flex>
-        <Flex alignItems="center">
-          {props.cluster.active ? (
-            <Label kind="success" ml={2} style={{ height: 'fit-content' }}>
-              active
-            </Label>
-          ) : null}
-          <ButtonIcon
-            mr="-10px"
-            style={{
-              visibility: isHovered ? 'visible' : 'hidden',
-              transition: 'none',
-            }}
-            ml={2}
-            title={`Log out from ${props.cluster.clusterName}`}
+      <Flex width="100%" justifyContent="space-between">
+        <WithIconItem
+          letter={getClusterLetter(props.cluster)}
+          color={profileColor}
+          title={props.cluster.name}
+          subtitle={props.cluster.loggedInUser?.name}
+        />
+        {props.onLogout && (
+          <ButtonText
+            intent="danger"
+            size="small"
+            className="logout"
+            css={`
+              visibility: hidden;
+              transition: none;
+            `}
+            p={1}
+            ml={4}
+            title={`Log out from ${props.cluster.name}`}
             onClick={e => {
               e.stopPropagation();
               props.onLogout();
             }}
           >
-            {/* Due to the icon shape it appears to be not centered, so a small margin is added */}
-            <Logout ml="2px" size="small" />
-          </ButtonIcon>
-        </Flex>
+            <Logout size="small" />
+          </ButtonText>
+        )}
       </Flex>
-    </ListItem>
+      {props.cluster.profileStatusError && (
+        <ProfileStatusError
+          error={props.cluster.profileStatusError}
+          // Align the error with the user icon.
+          css={`
+            margin-left: ${props => props.theme.space[2]}px;
+            gap: 11px;
```

Comment:
This seems to be off by 1px, 10px looks better IMHO. ;f

Reply:
Good eye! Changed to 10px.